src/ThreadTimer.hpp

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <thread>
+#include "Timer.hpp"
+
+class ThreadTimer : public Timer {
+public:
+    virtual void sleep_for(minutes duration) override{
```

Comment:
Is this 'virtual' needed here? (I am just asking)

Reply:
It is not